protobuf_serialization/writer.nim

```diff
@@ -173,12 +173,17 @@ proc writeValueInternal[T](stream: OutputStream, value: T) =
               {.fatal: "Encoding pragma specified yet no enoding matched. This should never happen.".}
 
           elif flattenedField is FixedTypes:
-            stream.writeFieldInternal(fieldNum, flattenedField, type(value), fieldName)
-
+            when sizeof(flattenedField) == 4:
+              stream.writeFieldInternal(fieldNum, Float32(flattenedField), type(value), fieldName)
```

Comment:
I'm having difficulties understanding why these `Float` types appear here. We are in the branch for `FixedTypes` which includes integer types as well. Why are we converting them to floats?

Reply:
sorry, that is my logical error, fixed.
clearly we need anti regression test suite.